app/src/androidTest/java/com/github/orkest/bluetooth/data/BluetoothServiceManagerTest.kt

```diff
@@ -35,14 +37,52 @@ class BluetoothServiceManagerTest {
 
     @Before
     fun setup() {
+        Constants.CURRENT_LOGGED_USER = "BLUETOOTH_TEST"
         bthServiceManager = BluetoothServiceManager(handler)
     }
 
     //==================TEST-CLIENT-CONNECTION=================
     @Test
-    fun connectToDevice() {
+    fun correctlyReceivesData() {
+        //Add data :to the inputStream of the socket
+
+        val testMsg:ByteArray = "Received".toByteArray()
+        val testDevice = TestDevice(testMsg)
+
+        msgReceived = ""
+        bthServiceManager.connectToDevice(testDevice)
+        Thread.sleep(1000)
+        assertEquals("Received", msgReceived)
     }
 
+    @Test
+    fun correctlySendsData() {
+        val testMsg:ByteArray = ByteArray(0)
+        val testDevice = TestDevice(testMsg)
+
+        msgReceived = ""
+        bthServiceManager.connectToDevice(testDevice)
+        Thread.sleep(1000)
+        assertEquals(Constants.CURRENT_LOGGED_USER, msgReceived)
+    }
+
+    @Test
+    fun cancelCorrectlyClosesSocket() {
+        val testMsg:ByteArray = ByteArray(0)
+        val testDevice = TestDevice(testMsg)
+
+        msgReceived = ""
+        bthServiceManager.connectToDevice(testDevice)
+        Thread.sleep(1000)
+        bthServiceManager.cancelConnections()
+        assertEquals(false, testDevice.socket.isConnected())
+        assertTrue(bthServiceManager.clientConnections.isEmpty())
+    }
+
```

Comment:
Instead of val testMsg:ByteArray = ByteArray(0) this is better in kotlin : val testMsg = byteArrayOf()

Why did you use Thread.sleep() instead of Synchronizing ?

Reply:
Ah yes I didn't know about the byteArrayOf() in Kotlin thank you.

Could you elaborate about the synchronizing, the threads I am launching for my tests are infinite loops waiting for connections, so I try to just wait a minimum of timefor the sokets to connect and the handler to send back the message